asynciterator.ts

```diff
@@ -1252,6 +1254,135 @@ function destinationFillBuffer<S>(this: InternalSource<S>) {
 }
 
 
+export abstract class SynchronousTransformIterator<S, D = S> extends AsyncIterator<D> {
+  private _destroySource: boolean;
+  protected constructor(protected _source: AsyncIterator<S>, protected options: { destroySource?: boolean } = {}) {
+    /* eslint-disable no-use-before-define */
+    super();
+    this._destroySource = options.destroySource !== false;
+    const cleanup = () => {
```

Comment:
The constructor definitely needs some cleaning up; it's very hard to understand what's going on with all of the helper functions.